weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx

```diff
@@ -127,7 +135,14 @@ export const CallsTable: FC<{
   );
 
   const effectiveFilter = useMemo(() => {
-    return {...filter, ...props.frozenFilter};
+    const workingFilter = {...filter, ...props.frozenFilter};
+    if (
+      !ALLOW_ALL_CALLS_UNFILTERED &&
+      !shouldForceNonTraceRootsOnly(workingFilter)
```

Comment:
I might be confused but I feel like this function is confusingly named - it doesn't exclude trace root calls does it? Seems like it just omits a root-only filter?

Reply:
Added some comments and renamed the function